packages/telemetry/src/index.ts

```diff
@@ -0,0 +1,241 @@
+import type { Span, SpanOptions, Tracer, Context } from '@opentelemetry/api'
+import type { BasicTracerProvider, SimpleSpanProcessor, BatchSpanProcessor, SpanExporter } from '@opentelemetry/sdk-trace-base'
+import type { DetectorSync } from '@opentelemetry/resources'
+
+import openTelemetry/*, { diag, DiagConsoleLogger, DiagLogLevel }*/ from '@opentelemetry/api'
+import { SemanticResourceAttributes } from '@opentelemetry/semantic-conventions'
+import { Resource, detectResourcesSync } from '@opentelemetry/resources'
+
+const types = ['child', 'root'] as const
+
+type AttachType = typeof types[number];
+
+type contextObject = {traceparent?: string}
+
+export type startSpanType = {
+  name: string
+  attachType?: AttachType
+  active?: boolean
+  opts?: SpanOptions
+}
+
+// Extend the span type to include span.name
+type SpanPlus = Span & { name: string }
+
+export type findActiveSpan =(element: SpanPlus, index: number) => boolean
+
+export class Telemetry {
+  tracer: Tracer
+  spans: {[key: string]: Span}
+  activeSpanQueue: Span[]
+  rootContext: Context | undefined
```

Comment:
Is this optional or are you expecting consumers to actually pass `undefined`? If optional, you can do:
```suggestion
  rootContext?: Context
```